pkg/engine/variables/regex/vars.go

```diff
@@ -11,7 +11,7 @@ var (
 	RegexReferences = regexp.MustCompile(`^\$\(.[^\ ]*\)|[^\\]\$\(.[^\ ]*\)`)
 
 	// RegexEscpReferences is the Regex for '\$(...)'
-	RegexEscpReferences = regexp.MustCompile(`\\\$\(.[^\ ]*\)`)
+	RegexEscpReferences = regexp.MustCompile(`\\\$\(.(?U)[^\ ]*\)`)
```

Comment:
I actually had to do some reading on `?U`  as it's a mode modifier which makes the the remainder of the expression ungreedy.

I am wondering what the difference to `\\\$\(.[^\ ]\)` which should (from my eyeball test) give the same result? I have not run the test cases though, just from a brief review.

Reply:
As you say `?U` make the pattern ungreedy. The greedy form will match the first variable found up to the end of the last variable found which will not allow to handle multiple variables substitution appropriately.
I've added a test case that show the current code doesn't work on a specific expression which is fixed by this PR.

Reply:
with 
`test = "\\$(NS_NAME)/\\$(NODE_NAME)/\\$(POD_NAME)"`

`FindAllString` returns 

with original regex

```
[
  \$(NS_NAME)/\$(NODE_NAME)/\$(POD_NAME)
]
```

with ungreedy regex

```
[
  \$(NS_NAME) 
  \$(NODE_NAME) 
  \$(POD_NAME)
]
```

another way could be

```suggestion
	RegexEscpReferences = regexp.MustCompile(`\\\$\(.[^\ )]*\)`)
```

Reply:
I agree that the current regex works, I was simply suggesting something more simple here without the usage of the relatively unknown`?U` flag.

Reply:
@MarcelMue Ok, I used the pattern proposed by @aodren to remove the ungreedy flag usage.